jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/core/extensions/PerMessageDeflateExtensionTest.java

```diff
@@ -435,6 +440,55 @@ public void testOutgoingPing() throws IOException
         ByteBufferAssert.assertEquals("Frame.payload", expected, actual.getPayload().slice());
     }
 
+    /**
+     * Outgoing Fragmented Message
+     * @throws IOException on test failure
+     */
+    @Test
+    public void testOutgoingFragmentedMessage() throws IOException, InterruptedException
+    {
+        PerMessageDeflateExtension ext = new PerMessageDeflateExtension();
+        ext.init(ExtensionConfig.parse("permessage-deflate"), bufferPool);
+
+        // Setup capture of outgoing frames
+        OutgoingFramesCapture capture = new OutgoingFramesCapture();
+
+        // Wire up stack
+        ext.setNextOutgoingFrames(capture);
+
+        Frame txtFrame = new Frame(OpCode.TEXT, false, "Hello ");
+        Frame con1Frame = new Frame(OpCode.CONTINUATION, false, "World");
+        Frame con2Frame = new Frame(OpCode.CONTINUATION, true, "!");
```

Comment:
Shouldn't you check for a protocol close below?

Reply:
this test is just creating the extension in isolation and sending frames through it which are passed to the `OutgoingFramesCapture`, so there is no protocol close going to happen

Reply:
So is there a test for the protocol exception?